src/index.ts

```diff
@@ -77,18 +77,44 @@ export function identity<B extends AnyBrand>(underlying: BaseOf<B>): B {
 }
 
 /**
- * Produces a `Brander<B>`, given a brand type `B`. This simply returns
- * `identity` but relies on type inference to give the return type the correct
- * type.
+ * Asserts that a value is the type `Output`.
+ */
+export type AssertionFunction<Input, Output extends Input> = (
+  value: Input,
+) => asserts value is Output;
+
+/**
+ * Produces a `Brander<B>`, given a brand type `B`. By default this returns
+ * `identity` and relies on type inference to give the return type the correct
+ * type. Optionally, `validator` can be used to assert on the value.
  *
- * @return `identity`
  * @example
  * ```
  * type UserId = Brand<number, 'user'>;
  * const UserId = make<UserId>();
  * const myUserId = UserId(42);
  * ```
+ * @example
+ * ```
+ * type UserId = Brand<number, 'user'>;
+ * const UserId = make<UserId>((value) => {
+ *   if (value <= 0) {
+ *     throw new Error(`Non-positive value: ${value}`);
+ *   }
+ * });
+ * UserId(42); // Ok
+ * UserId(-1); // Error: Non-positive value: -1
+ * ```
  */
-export function make<B extends AnyBrand>(): Brander<B> {
-  return identity;
+export function make<B extends AnyBrand>(
+  validator?: AssertionFunction<BaseOf<B>, B>,
+): Brander<B> {
+  if (!validator) {
+    return identity;
+  }
+
+  return (underlying: BaseOf<B>): B => {
+    (validator as Brander<B>)(underlying);
```

Comment:
> `as Brander<B>`

Without this, we get:

```plaintext
Assertions require every name in the call target to be declared with an explicit type annotation. ts(2775)
```

I've never seen that error before 😂 and don't know what to do to get around it. I couldn't figure it out from some GitHub searching & Googling and trying to add explicit type annotations around.

Reply:
Most search hits for the error message involve using arrow functions as an assertion function, but that doesn't seem like what's happening here.

Mysterious as it is, I do not consider this to be a blocker for merging.